src/main/groovy/eu/plumbr/plugins/jira/JiraPlugin.groovy

```diff
@@ -2,45 +2,132 @@ package eu.plumbr.plugins.jira
-  }
+    public static String earliestUnreleasedVersion(String jiraUser, String jiraPassword) {
+        final JerseyJiraRestClientFactory factory = new JerseyJiraRestClientFactory();
+        final URI jiraServerUri = new URI("https://plumbr.atlassian.net");
+        final JiraRestClient restClient = factory.createWithBasicHttpAuthentication(jiraServerUri, jiraUser, jiraPassword);
+        final NullProgressMonitor pm = new NullProgressMonitor();
+        final JiraProject p = restClient.getProjectClient().getProject("DTS", pm);
 
-  @Override
-  void apply(Project project) {
-    project.extensions.create('jira', JiraPluginExtension)
-    if (project.hasProperty('jiraUser')) {
-      project.jira.user = project.property('jiraUser')
+        p.getVersions().findAll { !it.isReleased() }.sort { it.releaseDate }.first().name
     }
-    if (project.hasProperty('jiraPassword')) {
-      project.jira.password = project.property('jiraPassword')
+
+    @Override
+    void apply(Project project) {
+        project.task('changeJiraStatusTask', type: JiraStatusTask)
+        project.extensions.create('jira', JiraPluginExtension)
+        if (project.hasProperty('jiraUser')) {
+            project.jira.user = project.property('jiraUser')
+        }
+        if (project.hasProperty('jiraPassword')) {
+            project.jira.password = project.property('jiraPassword')
+        }
+        if (project.hasProperty('artifactoryUser')) {
+            project.jira.artifactoryUser = project.property('artifactoryUser')
+        }
+        if (project.hasProperty('jiraPassword')) {
+            project.jira.artifactoryPassword = project.property('artifactoryPassword')
+        }
     }
-  }
 }
 
 class JiraPluginExtension {
-  String user
-  String password
-  String version
+    String user
+    String password
+    String version
+    String artifactoryUser
+    String artifactoryPassword
+
+    public String getVersion() {
+        if (version == null && user != null) {
+            version = JiraPlugin.earliestUnreleasedVersion(user, password)
+        }
+        version
+    }
+
+}
+
+class JiraStatusTask extends DefaultTask {
+
+    @Input
+    String buildNum
 
-  public String getVersion() {
-    if (version == null && user != null) {
-      version = JiraPlugin.earliestUnreleasedVersion(user, password)
+    @TaskAction
+    def executeStatusChange() {
+        changeJiraStatus(buildNum)
     }
-    version
-  }
 
+    public void changeJiraStatus(def buildNum) {
+        def restClient = new RESTClient("https://plumbr.artifactoryonline.com/plumbr/api/build/dashboard/${buildNum}")
+        restClient.auth.basic project.jira.artifactoryUser, project.jira.artifactoryPassword
+        def jsonResponse = restClient.get(contentType: ContentType.JSON).data
+        if (_isRelease(jsonResponse))
+            if (_getKeys(jsonResponse) != null)
+                _getKeys(jsonResponse).each { v -> _toClosedStatus(v) }
```

Comment:
You call `getKeys` method twice here. Why recalculate the same result 2 times? Extract to variable.